cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -169,6 +183,145 @@ protected void setMetricsEnabled(boolean isMetricsEnabled) {
+                JsonObject json = JsonObject.mapFrom(origin.get("fromConfigMap"));
+                String cmName = json.getString("name");
+                if (cmName == null) {
+                    log.warn("Field fromConfigMap must contain \"name\"");
+                } else {
+                    this.customCMname = cmName;
+                }
+            } else {
+                // validate all entries
+                for (Map.Entry entry: origin.entrySet()) {
+                    if (validLoggerFields.containsKey(entry.getKey())) {
+                        // correct logger
+                    } else {
+                        // incorrect logger
+                        log.warn(entry.getKey() + " is not valid logger");
+                    }
+                    if (entry.getKey().toString().contains("log4j.appender.CONSOLE")) {
+                        log.warn("You cannot set appender");
+                    }
+                    if ((Arrays.asList(validLoggerValues).contains(entry.getValue().toString().replaceAll(", CONSOLE", ""))) || (Arrays.asList(validLoggerValues).contains(entry.getValue()))) {
+                        // correct value
+                    } else {
+                        String tmpEntry = entry.getValue().toString();
+                        Pattern p = Pattern.compile("\\$\\{(.*)\\}, ([A-Z]+)");
+                        Matcher m = p.matcher(tmpEntry);
+
+                        String logger = "";
+                        String value = "";
+                        boolean regexMatch = false;
+                        while (m.find()) {
+                            logger = m.group(1);
+                            value = m.group(2);
+                            regexMatch = true;
+                        }
+                        if (regexMatch) {
+                            if (!validLoggerFields.containsKey(logger)) {
+                                log.warn(logger + " is not a valid logger");
+                            }
+                            if (!value.equals("CONSOLE")) {
+                                log.warn(value + " is not a valid value.");
+                            }
+                        } else {
+                            log.warn(entry.getValue() + " is not a valid value. Use one of " + Arrays.toString(validLoggerValues));
+                        }
+                    }
+                }
+                // update fields otherwise use default values
+                newSettings = getDefaultLogConfig();
+                for (Map.Entry<String, Object> entry : origin.entrySet()) {
+                    newSettings.replace(entry.getKey(), entry.getValue().toString());
+                }
+                this.logConfig = createPropertiesString(newSettings);
+            }
+        }
+    }
+
+    public String getLogConfigName() {
+        return logConfigName;
+    }
+
+    /**
+     * Sets name of config map, where logging configuration is stored
```

Comment:
"where logging configuration is stored", do you mean the CM which the user specifies, or the one which is managed by the CO?